### PR TITLE
feat: add support for data-based conditions on smart action trigger, approval requirement and authorization to approve

### PR DESCRIPTION
**Pull request by Thenkei**
…

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by Thenkei** [comment]
Awaiting `@forestadmin/forestadmin-client` newer version.

**Comment by Thenkei** [comment]
I've tested the workflow. It working smoothly on my end. :)

File: src/services/authorization/action-authorization.ts [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+    if (!rolesIdsGroupByConditions.length) {
+      return roleIdsAllowedToApproveWithoutConditions;
+    }
+
+    const [requestRecordsCount, ...conditionRecordsCounts] = await Promise.all([
+      ActionAuthorizationService.aggregateCountConditionIntersection({
+        ...recordsCounterParams,
+        excludesScope: true,
+      }, filterForCaller),
+      // eslint-disable-next-line max-len
```
Comment (Thenkei) [inline src/services/authorization/action-authorization.ts:254]: To remove

Comment (ghusse) [reply src/services/authorization/action-authorization.ts:254]: Nope, the formatter keeps formatting the line as one long line :'(

**Comment by ghusse** [comment]
Linked to CU-35d7gv6

**Review by ghusse — APPROVED** [review]
- ✅&nbsp;Code review
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
  - ✖️&nbsp; Integration tests
- ✅&nbsp;Manual tests
- ✅&nbsp;PR title
- ✅&nbsp;Destination branch
- ✅&nbsp;PR linked to the clickup task

**Pull request merged**